packages/mui-material/src/Select/SelectInput.js

```diff
@@ -400,6 +400,9 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
       if (value) {
         return selected;
       }
+      if (valueProp === '') {
```

Comment:
I don't follow this logic. An empty string is a valid value. Why exclude it?

Reply:
@oliviertassinari You are right. If the option's value has empty string and Select's initial value is empty string, then that option should be selected initially which does not happen with this change. Also we are not able to navigate with keys. I will have to take a look.

Reply:
The following diff works to not select first option when Select's value is empty string and also select the empty value option when available if Select's value is empty string:
```diff
diff --git a/packages/mui-material/src/Select/SelectInput.js b/packages/mui-material/src/Select/SelectInput.js
index 0ace49003c..e73389a8e6 100644
--- a/packages/mui-material/src/Select/SelectInput.js
+++ b/packages/mui-material/src/Select/SelectInput.js
@@ -397,12 +397,9 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
     }

     const isFirstSelectableElement = () => {
-      if (value) {
+      if (value !== undefined) {
         return selected;
       }
-      if (valueProp === '') {
-        return false;
-      }
       const firstSelectableElement = arr.find(
         (item) => item.props.value !== undefined && item.props.disabled !== true,
       );
:
```

But then the keyboard navigation does not work if first child is a `ListSubheader` (#20448). I think we have to revert #27299 which selects the first selectable option by default to enable keyboard navigation but it creates a bug when initial value is empty string (#34731) which this PR tried to resolve.

Let me know how to proceed.

Reply:
IMO the MenuList should recognize if the child it wants to focus is a ListSubheader and skip it. The code in https://github.com/mui/material-ui/blob/c616ce9552d03699008fe896c6602e2262e45cf3/packages/mui-material/src/MenuList/MenuList.js#L237-L251 assumes the child can be focusable, which is not right in case of ListSubheader.
This way we'll be able to revert #27299 and fix the problem at the source instead of patching the symptoms.

Reply:
@michaldudak ListSubheader or regular MenuItem that is disabled.